### ISSUE TITLE
Using the InAppBrowser.open({url: "https://test.com"}) gives {"code":"UNIMPLEMENTED"}

### ISSUE THREAD

**Issue by heblol** [Cap-go/capacitor-inappbrowser#211]
Hi there!

I just saw this fantastic plugin, however, it's not working for me? Trying to do the following:

```
await InAppBrowser.open({ url: "http://google.com" })
        .then((result) => {
          console.log("result opening browser", result);
        })
        .catch((error) => {
          console.error("error opening browser", error);
        });
```
Using the `openWebView` function gave the same error (`await InAppBrowser.openWebView(...)`)

However, it gives me the error:
`⚡️  [error] - error opening browser {"code":"UNIMPLEMENTED"}`

I tried no to await the `InAppBrowser.open(...)`, but that gave me no response at all.

Any idea what's going on?

iOS: (virtual) Iphone 16 - IOS 18.0
capacitorjs version: 6.1.2

using it with ReactJs

Full component:
```
import { Button, Card, CardContent, TextField } from "@mui/material";
import { Title } from "../../components";
import { useState } from "react";
import { InAppBrowser } from "@capgo/inappbrowser";
import { Capacitor } from "@capacitor/core";

export default function RecipeURLImport() {
  const [recipeURL, setRecipeURL] = useState<string>("");

  async function openBrowser() {
    console.log("clicked on open browser");
    if (Capacitor.isNativePlatform()) {
      console.log("trying to open the browser?");
      await InAppBrowser.open({ url: "http://google.com" })
        .then((result) => {
          console.log("result opening browser", result);
        })
        .catch((error) => {
          console.error("error opening browser", error);
        });
      console.log("after opening browser??");
      InAppBrowser.addListener("browserPageLoaded", () => {
        console.log("browserPageLoaded");
      });

      InAppBrowser.addListener("messageFromWebview", (event) => {
        console.log("messageFromWebview", event);
      });

      InAppBrowser.addListener("urlChangeEvent", (event) => {
        console.log("urlChangeEvent", event);
      });

      InAppBrowser.addListener("confirmBtnClicked", (event) => {
        console.log("confirmBtnClicked", event);
      });
    } else {
      console.log("trying to open it using the window.open");
      window.open("https://dansen.nl");
    }
  }

  return (
    <Card>
      <CardContent>
        <Title>Import recipe using an URL</Title>
        <Button
          onClick={() => {
            openBrowser();
          }}
        >
          Open browser
        </Button>
      </CardContent>
    </Card>
  );
}

```

Thank you in advance!

**Comment by riderx** [comment]
Hello did you cap sync ? it's seems weird like the native code didn't arrive in your app code

**Comment by nasaleanhorea** [comment]
I had the same issue , for me the problem was that i was using v5 for @capacitor/cli to when npx cap sync , which is why the plugin wasn't getting fully implemented in the native code -> hence the error. After upgrading to @capacitor/cli v6 , everything worked as expected on both iOS and Android.  With v5 it only worked on Android FYI.

**Comment by riderx** [comment]
Yes the plugin require capacitor 6, and now the latest require 7

**Comment by tempire** [comment]
I'm getting the same when run directly from xcode. When run on the simulator, I get this in the safari devtools:

> Error: "InAppBrowser" plugin is not implemented on ios

```javascript
import { InAppBrowser } from '@capgo/inappbrowser'
import { Capacitor } from "@capacitor/core"
console.log({isNative: Capacitor.isNativePlatform()})
const thing = await InAppBrowser.open('http://apache.org')
```

> capacitor sync ios
[capacitor] ✔ Copying web assets from dist to ios/App/App/public in 80.53ms
[capacitor] ✔ Creating capacitor.config.json in ios/App/App in 1.30ms
[capacitor] ✔ copy ios in 165.60ms
[capacitor] ✔ Updating iOS plugins in 1.93ms
[capacitor] ✔ Updating iOS native dependencies with pod install in 2.88s
[capacitor] [info] Found 8 Capacitor plugins for ios:
[capacitor]        @aparajita/capacitor-biometric-auth@8.0.2
[capacitor]        @capacitor-mlkit/barcode-scanning@7.0.0-next.3
[capacitor]        @capacitor/app@7.0.0
[capacitor]        @capacitor/browser@7.0.0
[capacitor]        @capacitor/haptics@7.0.0
[capacitor]        @capacitor/keyboard@7.0.0
[capacitor]        @capacitor/status-bar@7.0.0
[capacitor]        @capgo/inappbrowser@7.2.9
[capacitor] ✔ update ios in 2.93s
[capacitor] [info] Sync finished in 3.391s

**Comment by riderx** [comment]
I can reproduce, i think it's linked to capacitor 7 migration, fixing it

**Comment by tempire** [comment]
If it's relevant, I went through this tutorial on creating a plugin, and I get the same problem.

https://ionic.io/blog/creating-your-own-media-plugin-for-capacitor-an-ios-tutorial

**Comment by riderx** [comment]
Just deployed a fix, please try it ` 7.2.13` :)
PS: this tutorial is too old, look here it's better: https://capacitorjs.com/docs/plugins/creating-plugins

**Comment by tempire** [comment]
It works!

Was the fix just the addition of the bridged protocol in this commit? https://github.com/Cap-go/capacitor-inappbrowser/commit/78056b036d3c05e9c8085930ded5e6d89a3d37d0

I'll check out the newer tutorial - thanks.

**Comment by riderx** [comment]
yes i remove the .h and .m file to use the new protocole and added in main file function definition but forget to inerit the class
